app/app.py

```diff
@@ -197,10 +224,25 @@ def _create_payload(
         Dict[str, List[Dict[str, Any]]]: the payload to send in the correct format.
     """
     context = messages.get("context", DEFAULT_CONTEXT)
-    input_text = prompt_templates[st.session_state.prompt_template].format(
-        question=messages["prompt_query"], context=context
-    )
+    history: List[Dict[str, str]] = messages.get("history", [])
+    question = messages["prompt_query"]
+
+    if st.session_state.prompt_template == "conversational":
```

Comment:
Nit: `"conversational"` could be a static variable at the top of the file, similar with `"simple"`

Reply:
Not sure what that means in here. `conversational` is a key in `prompt_templates` dict. Reason for having line 230 is that the logic of `conversational` template is different to other template as it needs to use the conversation memory from the session state and build it as a string to pass it to the model.